tokens/api_server.go

```diff
@@ -0,0 +1,305 @@
+package tokens
+
+import (
+	"context"
+	"fmt"
+	log "github.com/sirupsen/logrus"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+)
+
+const (
+	defaultSecret          = "secret"
+	defaultTokenTTL        = "57600000"
+	defaultRefreshTokenTTL = "7200000"
+)
+
+type tokenAPIServer struct {
+	ctx          context.Context
+	client       *config.ManagementContext
+	tokensClient v3.TokenInterface
+}
+
+func newTokenAPIServer(ctx context.Context, mgmtCtx *config.ManagementContext) (*tokenAPIServer, error) {
+	if mgmtCtx == nil {
+		return nil, fmt.Errorf("Failed to build tokenAPIHandler, nil ManagementContext")
+	}
+	apiServer := &tokenAPIServer{
+		ctx:          ctx,
+		client:       mgmtCtx,
+		tokensClient: mgmtCtx.Management.Tokens(""),
+	}
+	return apiServer, nil
+}
+
+//CreateLoginToken will authenticate with provider and create a jwt token
+func (s *tokenAPIServer) createLoginToken(jsonInput v3.LoginInput) (v3.Token, int, error) {
+
+	log.Info("Create Token Invoked %v", jsonInput)
+	authenticated := true
+
+	/* Authenticate User
+		if provider != nil {
+		token, status, err := provider.GenerateToken(json)
+		if err != nil {
+			return model.TokenOutput{}, status, err
+		}
+	}*/
+
+	if authenticated {
+
+		if s.client != nil {
+
+			key, err := generateKey()
+			if err != nil {
+				log.Info("Failed to generate token key: %v", err)
+				return v3.Token{}, 0, fmt.Errorf("Failed to generate token key")
+			}
+
+			//check that there is no token with this key.
+			payload := make(map[string]interface{})
+			tokenValue, err := createTokenWithPayload(payload, defaultSecret)
```

Comment:
Did we discuss not needing a JWT? That was legacy that is no longer needed bc we will only ever return a a token id to the user. There is no need to store anything as a JWT.